src/Features/Tas/TasTools/CheckTool.cpp

```diff
@@ -79,13 +86,33 @@ static QAngle parseAng(const std::vector<std::string> &args, size_t idx) {
 	}
 }
 
+static void parseVel(const std::vector<std::string>& args, size_t idx, std::optional<float> &vel, std::optional<char> &veldir, std::optional<char> &velcomp) {
+	if (args.size() - idx < 1) {
+		throw TasParserException("Bad vel for check tool");
+	}
+	try {
+		vel = std::stoi(args[idx+2]);
+		veldir = args[idx].c_str()[0];
+		velcomp = args[idx+1].c_str()[0];
+
+		if (!((veldir.value() == 'x' || veldir.value() == 'y' || veldir.value() == 'z') && (velcomp.value() == '>' || velcomp.value() == '<'))) {
+			throw TasParserException("Bad args");
```

Comment:
This throw here will be caught immediately in the catch all `(...)` case which means that giving this a message of "Bad args" has no effect. Was this intended? You probably want to move the check out of the try-catch.